clang/lib/CIR/CodeGen/CIRGenExprCXX.cpp

```diff
@@ -901,17 +901,11 @@ void CIRGenFunction::emitCXXDeleteExpr(const CXXDeleteExpr *E) {
     return;
   }
 
-  // We might be deleting a pointer to array.  If so, GEP down to the
-  // first non-array element.
-  // (this assumes that A(*)[3][7] is converted to [3 x [7 x %A]]*)
-  if (DeleteTy->isConstantArrayType()) {
-    llvm_unreachable("NYI");
```

Comment:
Perhaps we should leave this around and just improve the comments so it's easier to follow later on, example:
```
  // In CodeGen:
  // We might be deleting a pointer to array.  If so, GEP down to the
  // first non-array element.
  // (this assumes that A(*)[3][7] is converted to [3 x [7 x %A]]*)
  // In CIRGen: we handle this differently because .... .. .. . .
  if (DeleteTy->isConstantArrayType()) {
    Ptr = Ptr;
  }
  ....
```

Reply:
Done